js/src/elements/star-rating.ts

```diff
@@ -267,24 +290,35 @@ export class StarRatingElement extends HTMLElement {
 		if (name === 'value') {
 			this.value = newValue;
 			this.updateStarFill(parseInt(newValue));
+			const outputElement = this.shadowRoot?.querySelector(
+				'.star-rating-output'
+			) as HTMLOutputElement;
+			if (target?.nextElementSibling) {
+				outputElement.textContent = target?.nextElementSibling?.textContent!.trimStart();
+			}
 		}
 		if (name === 'disabled') {
 			const fieldset = this.shadowRoot?.querySelector('fieldset') as HTMLFieldSetElement;
 			fieldset.disabled = newValue !== null;
 		}
+		if (name === 'required') {
+			const input = this.shadowRoot?.querySelectorAll('input[type="radio"]')[0] as HTMLInputElement;
+			input.required = newValue !== null;
+			const requiredIndicator = document.createElement('span');
+			requiredIndicator.setAttribute('part', 'required-indicator');
+			requiredIndicator.ariaHidden = 'true';
+			const lastStar = this.shadowRoot?.querySelectorAll('label')[4] as HTMLLabelElement;
+			lastStar.insertAdjacentElement('afterend', requiredIndicator);
+		}
```

Comment:
Updated

Reply:
I meant in the context of a form with other elements using the new .field class and structure, it would probably make sense to display the star rating in the same way:

![image](https://user-images.githubusercontent.com/1306747/175662174-d8269d46-e5c2-4364-9839-4ea0e0ec4b5b.png)

```diff
  <div class="field">
+   <div class="field-label">Star rating<span class="required-indicator"></span></div>
    <div class="field-body">
      <star-rating id="star-rating-1" name="star-rating-1" required>
-     <legend slot="legend">Star rating</legend>
+     <legend slot="legend" class="visually-hidden">Star rating<span class="required-indicator"></span></legend>
      <span slot="label-1">Terrible</span>
      <span slot="label-2">Poor</span>
      <span slot="label-3">Fair</span>
      <span slot="label-4">Good</span>
      <span slot="label-5">Great</span>
      </star-rating>
    </div>
  </div>
```

Reply:
Updated